sotrplib/sources/core.py

```diff
@@ -99,8 +102,61 @@ def search(
         return [], []
 
 
+@dataclass
+class BlindSearchParameters:
+    """
+    Parameters for blind source searching using photutils.
+    """
+
+    sigma_threshold: float = 5.0
+    minimum_separation: list[u.Quantity] = field(
+        default_factory=lambda: [u.Quantity(0.5, "arcmin")]
+    )
+    sigma_threshold_for_minimum_separation: list[float] = field(
+        default_factory=lambda: [3.0]
+    )
+
+
 class SigmaClipBlindSearch(BlindSearchProvider):
+    def __init__(
+        self,
+        input_map: ProcessableMap,
```

Comment:
The input map shouldn't be part of the initialization process for the class. Remember, with Dependency Injection we're looking to 'set up' the search (i.e. configure it) with the initialization, and then 'apply' it to many maps with the `search` function.